runtime/src/iree/hal/drivers/metal/builtin_executables.h

```diff
@@ -17,13 +17,22 @@
 extern "C" {
 #endif  // __cplusplus
 
+// Object and launch parameters for a compute function.
+typedef struct iree_hal_metal_builtin_pipeline_t {
+  id<MTLComputePipelineState> pipeline_state;
+  IREE_TRACE(iree_hal_metal_source_location_t source_location;)
+} iree_hal_metal_builtin_pipeline_t;
+
 typedef struct iree_hal_metal_builtin_executable_t {
   iree_allocator_t host_allocator;
 
+  // Compiled MTLLibrary instances containing the builtin kernels.
+  NSArray<id<MTLLibrary>>* libraries;
+
   // The number of entry points in this builtin executable.
-  iree_host_size_t entry_point_count;
+  iree_host_size_t pipeline_count;
   // THe list of entry points, pointing to the end of the struct allocation.
```

Comment:
Unchanged line, but spotted typo
```suggestion
  // The list of entry points, pointing to the end of the struct allocation.
```